framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -949,7 +937,16 @@ protected boolean visitIntersectionSupertype(
         if (visitHistory.contains(subtype, supertype, currentTop)) {
             return true;
         }
-        boolean result = isSubtypeOfAll(subtype, supertype.directSuperTypes());
+        boolean result = true;
+        for (AnnotatedTypeMirror aSupertype : supertype.directSuperTypes()) {
+            if (TypesUtils.isErasedSubtype(
```

Comment:
A comment explaining why there is this check for `isErasedSubtype` would be useful.